sys/net/routing/rpl/rpl_dodag.c

```diff
@@ -238,16 +234,16 @@ void rpl_delete_worst_parent(void)
 
 }
 
-void rpl_delete_all_parents(void)
+void rpl_delete_all_parents(rpl_dodag_t *dodag)
 {
-    rpl_dodag_t *my_dodag = rpl_get_my_dodag();
-
-    if (my_dodag != NULL) {
-        my_dodag->my_preferred_parent = NULL;
-    }
 
+    dodag->my_preferred_parent = NULL;
     for (int i = 0; i < RPL_MAX_PARENTS; i++) {
-        memset(&parents[i], 0, sizeof(parents[i]));
+        if (dodag->instance->id == parents[i].dodag->instance->id &&
+                !memcmp(&dodag->dodag_id,
+                    &parents[i].dodag->dodag_id, sizeof(ipv6_addr_t))) {
+             memset(&parents[i], 0, sizeof(parents[i]));
```

Comment:
I think this must be `rpl_delete_parent(&parents[i])` to additionally delete the `dodag->my_preferred_parent` if its the same one